UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/WebhookController.cs

```diff
@@ -44,10 +48,38 @@ public virtual async Task<IActionResult> Post()
 
-            _logger.LogInformation(EventIds.ENSWebhookRequestStart.ToEventId(), "External notification service webhook request started for event:{eventGridEvent} and _X-Correlation-ID:{correlationId}.", JsonConvert.SerializeObject(eventGridEvent), GetCurrentCorrelationId());
+            if (customEventGridEvent != null)
+            {
+                var processor = _webhookService.GetProcessor(customEventGridEvent.Type);
+
+                if (processor != null)
+                {
+                    ExternalNotificationServiceProcessResponse result = await processor.Process(customEventGridEvent, GetCurrentCorrelationId());
+
+                    if (result.Errors != null && result.Errors.Count > 0)
+                    {
+                        _logger.LogInformation(EventIds.ENSWebhookRequestInvalidEventPayload.ToEventId(), "External notification service webhook request is failed due to invalid event payload for subject:{subject}, businessUnit:{businessUnit} and _X-Correlation-ID:{correlationId}.", customEventGridEvent.Subject, result.BusinessUnit, GetCurrentCorrelationId());
+                        _logger.LogInformation(EventIds.ENSWebhookRequestCompleted.ToEventId(), "External notification service webhook request completed for subject:{subject} and _X-Correlation-ID:{correlationId}.", customEventGridEvent.Subject, GetCurrentCorrelationId());
+                        
+                        return BuildOkRequestErrorResponse(result.Errors);
+                    }
+                }
+                else 
+                {
+                    _logger.LogInformation(EventIds.ENSWebhookRequestTypeNotMatch.ToEventId(), "External notification service webhook request is failed due to event type not matched with event processor for subject:{subject}, type:{type} and _X-Correlation-ID:{correlationId}.", customEventGridEvent.Subject, customEventGridEvent.Type, GetCurrentCorrelationId());
+                }
+
+                _logger.LogInformation(EventIds.ENSWebhookRequestCompleted.ToEventId(), "External notification service webhook request completed for subject:{subject} and _X-Correlation-ID:{correlationId}.", customEventGridEvent.Subject, GetCurrentCorrelationId());
+            }
+            else
+            {
+                _logger.LogInformation(EventIds.ENSWebhookRequestWithNullEventPayload.ToEventId(), "External notification service webhook request is failed due to null event payload for _X-Correlation-ID:{correlationId}.", GetCurrentCorrelationId());
+                _logger.LogInformation(EventIds.ENSWebhookRequestCompleted.ToEventId(), "External notification service webhook request completed for _X-Correlation-ID:{correlationId}.", GetCurrentCorrelationId());
+            }
```

Comment:
This looks an unreachable code, JsonConvert.DeserializeObject won't return `null` any time (it may throw exception though). Please can we revisit this block?

Reply:
Thanks for highlighting, I have made the necessary changes to this block.